api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/ParameterNameConverter.scala

```diff
@@ -14,21 +14,24 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package ai.h2o.sparkling.ml.algos
 
-import ai.h2o.sparkling.ml.params.H2ODeepLearningParams
-import ai.h2o.sparkling.ml.utils.H2OParamsReadable
-import hex.deeplearning.DeepLearningModel.DeepLearningParameters
-import org.apache.spark.ml.util._
+package ai.h2o.sparkling.api.generation.common
 
-/**
-  * H2O DeepLearning algorithm exposed via Spark ML pipelines.
-  */
-class H2ODeepLearning(override val uid: String)
-  extends H2OSupervisedAlgorithm[DeepLearningParameters]
-  with H2ODeepLearningParams {
+object ParameterNameConverter {
+  val h2oToSWExceptions: Map[String, String] = Map(
+    "lambda" -> "lambdaValue",
+    "alpha" -> "alphaValue",
+    "colsample_bylevel" -> "colSampleByLevel",
+    "colsample_bytree" -> "colSampleByTree")
 
-  def this() = this(Identifiable.randomUID(classOf[H2ODeepLearning].getSimpleName))
-}
+  val conversionRules: Map[String, String] = Map("Column" -> "Col")
```

Comment:
If we have this conversion rule, why `folds_col` and similar params are mentioned also in IgnoredColumns class?

Reply:
`folds_col`, `weights_column`, `response_column` and others are ignored since they are already implemented in parent traits (e.g. `H2OAlgoSupervisedParams`, `H2OCommonSupervisedParams`). These traits are also used by grid search and AutoML. Maybe once parameters for Automl and GridSearch will be generated, there will be a room for simplyfing these classes.

Back to your question, this rule is also used for other parameters like `ignored_columns` -> `ignoredCols`

Reply:
I see, once grid & automl params are generated as well, we can remove this hiearchy and generate as well. That will be nice next step.

Thanks for explanation!